src/LoRa.cpp

```diff
@@ -582,7 +582,31 @@ void LoRaClass::setOCP(uint8_t mA)
 
 byte LoRaClass::random()
 {
-  return readRegister(REG_RSSI_WIDEBAND);
+  uint8_t currMode = readRegister(REG_OP_MODE);
+  uint8_t retVal = 0;
+  void (*_prevOnReceive)(int) = NULL;
+  
+  while(isTransmitting()) yield();
+
+  //We need to be listening to radio-traffic in order to generate random numbers
+  if(currMode != (MODE_LONG_RANGE_MODE | MODE_RX_CONTINUOUS)){
+#ifndef ARDUINO_SAMD_MKRWAN1300
```

Comment:
I'm generally oppose to these kinds of checks, since this PR was originally submit, 'support' has been added for the 1310 board(sorry this hasn't got attention...) and doesn't even begin to handle many other boards out there. Can this check instead be performed against `_onReceive` itself?